backend/excel_util.py

```diff
@@ -25,48 +81,41 @@ def election_to_xlsx(election, filename):
     xtd_final_shares = find_xtd_shares([xtd_final_votes])[0]
 
     workbook = xlsxwriter.Workbook(filename)
+    fmt = prepare_formats(workbook)
     worksheet = workbook.add_worksheet()
-    cell_format = workbook.add_format()
-    cell_format.set_align('right')
-    share_format = workbook.add_format()
-    share_format.set_num_format('0.0%')
-    h_format = workbook.add_format()
-    h_format.set_align('center')
-    h_format.set_bold()
-    h_format.set_font_size(14)
 
     worksheet.set_column('B:B', 20)
 
-    def write_matrix(worksheet, startrow, startcol, matrix, cformat):
-        for c in range(len(matrix)):
-            for p in range(len(matrix[c])):
-                if matrix[c][p] != 0:
-                    try:
-                        worksheet.write(startrow+c, startcol+p, matrix[c][p],
-                                        cformat[c])
-                    except TypeError:
-                        worksheet.write(startrow+c, startcol+p, matrix[c][p],
-                                        cformat)
-
     def draw_block(worksheet, row, col,
         heading, xheaders, yheaders,
         matrix,
         topleft="Constituency",
-        cformat=cell_format
+        cformat=fmt["cell"]
     ):
         if heading.endswith("shares"):
-            cformat = share_format
+            cformat = fmt["share"]
         worksheet.merge_range(
             row, col+1,
             row, col+len(xheaders),
-            heading, h_format
+            heading, fmt["h"]
         )
-        worksheet.write(row+1, col, topleft, cell_format)
-        worksheet.write_row(row+1, col+1, xheaders, cell_format)
-        worksheet.write_column(row+2, col, yheaders, cell_format)
+        worksheet.write(row+1, col, topleft, fmt["cell"])
+        worksheet.write_row(row+1, col+1, xheaders, fmt["cell"])
+        worksheet.write_column(row+2, col, yheaders, fmt["cell"])
         write_matrix(worksheet, row+2, col+1, matrix, cformat)
 
     startcol = 1
+
+    toprow=0
+    c1=1
+    left_span=2
+    c2=c1+left_span
+    worksheet.merge_range(toprow,c1,toprow,c2-1,"Date:",fmt["basic_h"])
+    worksheet.merge_range(toprow,c2,toprow,c2+1,datetime.now(),fmt["time"])
+    toprow+=1
+    worksheet.merge_range(toprow,c1,toprow,c2-1,"Test name:",fmt["basic_h"])
+    worksheet.write(toprow,c2,"My electoral system",fmt["basic"])
```

Comment:
TODO: Arrange for name of vote table to be accessible through election object.

Reply:
Will be addressed in next pull request

Reply:
As promised:
https://github.com/smari/voting/pull/73/commits/2dc30b868c970b069f34a96df60b2c7f9212da26
https://github.com/smari/voting/pull/73/commits/26dd0a351afd310934f56659a9e8414d2f682b2a